code/game/objects/items/candle.dm

```diff
@@ -76,3 +72,10 @@
 	if(lit)
 		return source_temperature
 	return 0
+
+/obj/item/candle/Crossed(var/obj/Proj)
+	if(isbeam(Proj))
+		//Can also be an obj/effect, but they both have the damage var
```

Comment:
This is some oldcoder-tier hack. You might need to use if(Proj?:damage && Proj?:damage !=0) instead. ?: is one of lummox's new null-conditional operators, but i'm not sure how haram is to use what is pretty much a new runtime operator.

Reply:
i can separate them out to an isbeam and isbeameffect function, but i feel that would put us back where we started

Reply:
Damian said it's fine to keep it as it is, since isbeam() is filtering the object.

Reply:
@jknpj I don't think those operators do what you think they do
This is "fine"
By which I mean it really isn't but there isn't a better alternative thanks to BYOND